catalyst/loggers/comet.py

```diff
@@ -0,0 +1,264 @@
+            else:
+                print("A project_name has not been set. Using default project name 'general' instead.")
+        else:
+            self.project_name = project_name
+
+        self.logging_disabled = os.getenv('COMET_AUTO_LOG_DISABLE', False)
+        self.workspace = workspace
+        self.api_key = api_key
+        self.comet_mode = os.getenv('COMET_MODE', 'online')
+
+        if experiment is None:
+            try:
+                if api_key is None:
+                    parser = configparser.ConfigParser()
+                    enviornment_api_key = os.environ.get('COMET_API_KEY')
+                    try:
+                        parser.read(CONFIG_FILE_PATH)
+                        config_file_api_key = parser.get("comet", "api_key")
+                    except:
+                        config_file_api_key = None
+                    if config_file_api_key:
+                        self.api_key = config_file_api_key
+                        print("Using the API key stored in the comet config file.")
+                    elif enviornment_api_key:
+                        self.api_key = enviornment_api_key
+                        print("Using the API key stored in the COMET_API_KEY' enviornment variable.")
+
+                if self.comet_mode == 'offline':
+                    print("Starting an Offline Experiment")
+                    self.offline_directory = os.getenv('COMET_OFFLINE_DIRECTORY')
+                    self.experiment = comet_ml.OfflineExperiment(
+                        project_name=self.project_name, workspace=self.workspace, offline_directory=self.offline_directory, disabled=self.logging_disabled)
+                else:
+                    self.experiment = comet_ml.Experiment(
+                        project_name=self.project_name, api_key=self.api_key, workspace=self.workspace, disabled=self.logging_disabled
+                    )
+
+            except BaseException as e:
+                experiment = None
+                print(e)
+        else:
+            self.experiment = experiment
+
+        if tags:
+            for tag in tags:
+                self.experiment.add_tag(tag)
+
+    def log_metrics(
+        self,
+        metrics: Dict[str, float],
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        key_parameters = [stage_key, loader_key, scope]
```

Comment:
I think in order to to be more explicit about how we're using these arguments please change this variable to `prefix_parameters`